parastell/magnet_coils.py

```diff
@@ -436,6 +409,83 @@ def create_magnet(self):
         shell = cq.Shell.makeShell(face_list)
         self.solid = cq.Solid.makeSolid(shell)
 
+    def in_toroidal_extent(self, lower_bound, upper_bound):
+        """Determines if the coil lies within a given toroidal angular extent,
+        based on filament coordinates.
+
+        Arguments:
+            lower_bound (float): lower bound of toroidal extent [rad].
+            upper_bound (float): upper bound of toroidal extent [rad].
+
+        Returns:
+            in_toroidal_extent (bool): flag to indicate whether coil lies
+                within toroidal bounds.
+        """
+        # Compute toroidal angle of each point in filament
+        toroidal_angles = np.arctan2(self._coords[:, 1], self._coords[:, 0])
+        # Ensure angles are positive
+        toroidal_angles = (toroidal_angles + 2 * np.pi) % (2 * np.pi)
+        # Compute bounds of toroidal extent of filament
+        min_tor_ang = np.min(toroidal_angles)
+        max_tor_ang = np.max(toroidal_angles)
+
+        # Determine if filament toroidal extent overlaps with that of model
+        if (min_tor_ang >= lower_bound or min_tor_ang <= upper_bound) or (
+            max_tor_ang >= lower_bound or max_tor_ang <= upper_bound
+        ):
+            in_toroidal_extent = True
+        else:
+            in_toroidal_extent = False
```

Comment:
I tried this before and, unfortunately, because of the way we need to define `lower_bound` and `upper_bound` in `MagnetSet._filter_coils`, we need an `or` operator between comparisons and not an `and` operator (the chain acts as an `and`). It's difficult to explain in a single comment but it has to do with which range toroidal angles are defined on and the edge case that the model's toroidal extent is > 180 degrees. There may be a better way of defining lower and upper bounds to check against that allows comparison chaining, but I can't think of one.